src/scroll-timeline-css-parser.js

```diff
@@ -0,0 +1,381 @@
+      if (this.lookAhead("/*", p)) {
+        while (this.lookAhead("/*", p)) {
+          this.eatComment(p);
+          this.eatWhitespace(p);
+        }
+        continue;
+      }
+
+      if (this.lookAhead("@scroll-timeline", p)) {
+        const { scrollTimeline, startIndex, endIndex } = this.parseScrollTimeline(p);
+        if (firstPass) this.scrollTimelineOptions.set(scrollTimeline.name, scrollTimeline);
+      } else {
+        const rule = this.parseQualifiedRule(p);
+        if (!rule) continue;
+        if (firstPass) {
+          this.extractAndSaveKeyframeName(rule.selector);
+        } else {
+          this.handleScrollTimelineProps(rule, p);
+        }
+      }
+    }
+
+    // If this sheet has no srcURL (like from a <style> tag), we are done.
+    // TODO: Otherwise, we have to find `url()` functions and resolve
+    // relative and path-absolute URLs to absolute URLs.
+    return p.sheetSrc;
+  }
+
+  getScrollTimelineName(animationName, target) {
+    for (let i = this.cssRulesWithTimelineName.length - 1; i >= 0; i--) {
+      const current = this.cssRulesWithTimelineName[i];
+      if (target.matches(current.selector)) {
+        if (!current['animation-name'] || current['animation-name'] == animationName) {
+          return current['animation-timeline'];
+        }
+      }
+    }
+
+    return null;
+  }
+
+  parseScrollTimeline(p) {
+    const startIndex = p.index;
+    this.assertString(p, "@scroll-timeline");
+    this.eatWhitespace(p);
+    let name = this.parseIdentifier(p);
+    this.eatWhitespace(p);
+    this.assertString(p, "{"); // eats {
+    this.eatWhitespace(p);
+
+    let scrollTimeline = {
+      name: name,
+      source: "auto",
+      orientation: undefined,
+      'scroll-offsets': undefined
+    };
+
+    while (this.peek(p) !== "}") {
+      const property = this.parseIdentifier(p);
+      this.eatWhitespace(p);
+      this.assertString(p, ":");
+      this.eatWhitespace(p);
+      scrollTimeline[property] = this.removeEnclosingDoubleQuotes(this.eatUntil(";", p));
+      this.assertString(p, ";");
+      this.eatWhitespace(p);
+    }
+
+    this.assertString(p, "}");
+    const endIndex = p.index;
+    this.eatWhitespace(p);
+    return {
+      scrollTimeline,
+      startIndex,
+      endIndex,
+    };
+  }
+
+  handleScrollTimelineProps(rule, p) {
+    // The animation-timeline property may not be used in keyframes
+    if (rule.selector.includes("@keyframes")) {
+      return;
+    }
+
+    // TODO is it enough to check with "includes()"
+    const hasAnimationName = rule.block.contents.includes("animation-name:");
+    const hasAnimationTimeline = rule.block.contents.includes("animation-timeline:");
+    const hasAnimation = rule.block.contents.includes("animation:");
+
+    let timelineNames = [];
+    let animationNames = [];
+
+    if (hasAnimationTimeline) {
+      timelineNames = this.extract(rule.block.contents, RegexMatcher.ANIMATION_TIMELINE)
```

Comment:
Code duplication.  Add a helper method. i.e.

extractMatches(regex) {
   return this.extract(...).split(...).map(...);
}